lib/generators/deimos/schema_class/templates/schema_record.rb.tt

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# autogenerated by Deimos, do not modify
+module Deimos
+  # :nodoc:
+  class <%= schema_classname(@current_schema) %> < SchemaRecord
+<%- fields.each do |field| -%>
+    # @return [<%= deimos_field_type(field) %>]
+    attr_accessor :<%= field.name %>
+<% end -%>
+<%- unless @current_schema.name =~ /\-key$/ -%>
+    # @return [Object] An optional payload key
+    attr_accessor :payload_key
+<% end -%>
+
+    # @override
+    def initialize(**kwargs)
+      super()
+      args = kwargs.with_indifferent_access
+<%- fields.each do |field| -%>
+      <%= "@#{field.name} = args[:#{field.name}]" %>
+<% end -%>
+      @payload_key = args[:payload_key]
+    end
+
+    # @override
+    def self.initialize_from_payload(payload)
```

Comment:
I don't really understand why this is a different method from `initialize`? A hash and a set of keyword args are essentially the same thing?

Reply:
The only difference is that it's handling the hashes recursively and initializing sub-records/enums/maps/arrays whereas `initialize` would accept the already formatted values. I think i need to write an example of how this would be used

Reply:
Looking into handling recursive initialization via. the `initialize` method